scipy/optimize/tests/test_optimize.py

```diff
@@ -526,6 +526,37 @@ def test_ncg_hessp(self):
                         atol=1e-6, rtol=1e-7)
 
 
+def test_maxfev_test():
+    rng = np.random.default_rng(271707100830272976862395227613146332411)
+
+    def cost(x):
+        return rng.random(1) * 1000  # never converged problem
+
+    for imaxfev in [1, 10, 50]:
+        for method in ['Powell']:  # TODO: extend to more methods
+            result = optimize.minimize(cost, rng.random(10),
+                                       method=method,
+                                       options={'maxfev': imaxfev})
+            assert result["nfev"] == imaxfev
+
+
+def test_wrap_scalar_function_with_validation():
+
+    def func(x):
+        return x
+
+    fcalls, func = optimize.optimize.\
+        _wrap_scalar_function_with_validation(func, np.asarray(1), 5)
+
+    for i in range(5):
+        func(np.asarray(i))
+        assert fcalls[0] == i+1
+
+    msg = "Too many function calls"
+    with assert_raises(optimize.optimize._MaxFuncCallError, match=msg):
+        func(np.asarray(i))  # exceeded maximum function call
+
+
```

Comment:
Something like this, I will let you commit here if you agree with the suggestion.
```suggestion
    with assert_raises(optimize.optimize._MaxFuncCallError, match=msg):
        func(np.asarray(i))

    match = ("The user-provided "
             "objective function must "
             "return a scalar value.")
     with assert_raises(ValueError, match=msg):
         func(np.array([1, 1]))

```